### DISCUSSION TITLE
warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]

### DISCUSSION THREAD

**Discussion by Annisa1020** [redis/hiredis#1082] [Q&A]

**Comment by michael-grunder** [comment]
Hi, thanks for the question.

Can you provide more details about the exact compiler you're using where you see this error?  I don't see it with `gcc 13`, `clang-12` or `clang-14`.

I think it may be a false positive warning, but regardless it's actually in libev, not hiredis:
https://github.com/enki/libev/blob/93823e6ca699df195a6c7b8bfa6006ec40ee0003/ev.h#L722

And doing a little further research, it does appear to be a false positive.  I'll post a thread on the matter for everyone's enjoyment.
 
http://lists.schmorp.de/pipermail/libev/2010q1/000933.html
